lib/datadog/tracing/contrib/active_support/notifications/subscription.rb

```diff
@@ -11,12 +11,19 @@ class Subscription
               :span_name,
               :options
 
-            def initialize(span_name, options, &block)
-              raise ArgumentError, 'Must be given a block!' unless block
+            # @param span_name [String] the operation name for the span
+            # @param options [Hash] options to pass during span creation
+            # @param on_start [Proc] a block to run when the event is fired,
+            #   might not include all required information in the `payload` argument.
+            # @param on_finish [Proc] a block to run when the event has finished processing,
+            #   possibly including more information in the `payload` argument.
+            def initialize(span_name, options, on_start: nil, on_finish: nil)
```

Comment:
Would you agree that the interface is more streamlined with `on_start` and `on_finish` being the keys from the `options` hash? 

Oh, this comment also applies to other methods that accept `on_start`, `on_finish` and `options`.

Reply:
`options` are span options, and passed to `span = Tracing.trace(@span_name, **@options)`, we can't pass `on_start` to `Tracing.trace`.

I can rename options to `span_options` to make it clear.

Reply:
I renamed it to `span_options`.